pkg/apis/app/v1/kieapp_types.go

```diff
@@ -108,16 +108,29 @@ type KieAppObjects struct {
 	SmartRouter *KieAppObject `json:"smartRouter,omitempty"`
 }
 
+// PrometheusExtDisabledChoiceType Type used to define the user choice regarding prometheus ext use
+type PrometheusExtDisabledChoiceType string
+
+const (
+	// NotDefined if user didn't set the option
+	NotDefined PrometheusExtDisabledChoiceType = ""
+	// Enabled extension should be enabled
+	Enabled PrometheusExtDisabledChoiceType = "Enabled"
+	// Disabled extension should be disabled
+	Disabled PrometheusExtDisabledChoiceType = "Disabled"
+)
+
 // KieServerSet KIE Server configuration for a single set, or for multiple sets if deployments is set to >1
 type KieServerSet struct {
-	Deployments         *int                    `json:"deployments,omitempty"` // Number of KieServer DeploymentConfigs (defaults to 1)
-	Name                string                  `json:"name,omitempty"`
-	ID                  string                  `json:"id,omitempty"`
-	From                *corev1.ObjectReference `json:"from,omitempty"`
-	Build               *KieAppBuildObject      `json:"build,omitempty"` // S2I Build configuration
-	SecuredKieAppObject `json:",inline"`
-	Database            *DatabaseObject  `json:"database,omitempty"`
-	Jms                 *KieAppJmsObject `json:"jms,omitempty"`
+	Deployments          *int                    `json:"deployments,omitempty"` // Number of KieServer DeploymentConfigs (defaults to 1)
+	Name                 string                  `json:"name,omitempty"`
+	ID                   string                  `json:"id,omitempty"`
+	From                 *corev1.ObjectReference `json:"from,omitempty"`
+	Build                *KieAppBuildObject      `json:"build,omitempty"` // S2I Build configuration
+	SecuredKieAppObject  `json:",inline"`
+	Database             *DatabaseObject                 `json:"database,omitempty"`
+	Jms                  *KieAppJmsObject                `json:"jms,omitempty"`
+	DisablePrometheusExt PrometheusExtDisabledChoiceType `json:"disablePrometheusExt,omitempty"`
```

Comment:
Do we really need 3 modes for this (i.e., enabled, disabled, and unspecified)?

If we don't need three modes, let's make it a bool that is disabled by default, unless enabled.

If we need three modes, create a struct for it that has an enabled as a bool

Reply:
The original design was a checkbox. But the requirement has 3 states: undefined, true or false. If undefined the env variable shouldn't exist. That's why I opt for adding a dropdown instead a checkbox. But, I have to agree that this makes little sense.

@errantepiphany could you clarify? Don't you think that this requirement is a bit odd?

> The way the templates and code are now, the system property is only set IF a value for the PROMETHEUS_SERVER_EXT_DISABLED parameter is set. We do not default it to anything.

It's a boolean value, so defaulting to FALSE (unchecked), would make sense for this scenario since the prometheus extension would be enabled by default. Doing this in templates would be different from the APB and Operator since they have UI.

Reply:
@ricardozanini I'm okay with the checkbox, and defaulting to unchecked (disabled = false) value.